### PR TITLE
gertty: init at 1.6.0

### PR DESCRIPTION
**Pull request by petrodavi**
###### Motivation for this change

Add gertty package to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/tools/gertty/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{ python3, python3Packages, stdenv }:
```
Comment (knedlsepp) [inline pkgs/development/tools/gertty/default.nix]: - You are using both `python3Packages` and `python3.pkgs`, which are equivalent, you can get rid of one of them.
- You can directly make use of `lib` if you only need that one instead of `stdenv.lib`

File: pkgs/development/tools/gertty/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  src = python3.pkgs.fetchPypi {
+    inherit pname version;
+    sha256 = "1pvpca59z5rcir7lz2yimwiqjvd8yhnwjskxna2bslyfwv996d8w";
+  };
+
+  doCheck = true;
+
+  # XXX Missing maintainers?
```
Comment (knedlsepp) [inline pkgs/development/tools/gertty/default.nix]: According to the contributions guidelines newly submitted packages should have a maintainer: https://nixos.org/nixpkgs/manual/#reviewing-contributions-package-updates
Feel free to add yourself as maintainer in maintainers/maintainer-list.nix. This would mainly involve you receiving notifications on pull-requests about your package or (if these will be re-enabled) e-mail notifications on failing [hydra builds](https://hydra.nixos.org/job/nixpkgs).

File: pkgs/development/tools/gertty/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{ python3, python3Packages, stdenv }:
+
+python3.pkgs.buildPythonPackage rec {
```
Comment (knedlsepp) [inline pkgs/development/tools/gertty/default.nix]: If you only want to expose the `gertty` binary, please use `buildPythonPackage`, if you also want to expose gertty as python-library, please move the package to `python-modules`. (See 15.17.2.2.2 in the manual: https://nixos.org/nixpkgs/manual/)

File: pkgs/development/tools/gertty/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  pname = "gertty";
+  version = "1.6.0";
+
+  nativeBuildInputs = with python3Packages; [
```
Comment (knedlsepp) [inline pkgs/development/tools/gertty/default.nix]: It's unlikely that these packages are needed in the nativeBuildInputs. You can find in `15.17.2.2.1.1. buildPythonPackage parameters` (Somewhere below here: https://nixos.org/nixpkgs/manual/#reference ) that in the case of python:

nativeBuildInputs ? []: Build-time only dependencies. Typically executables as well as the items listed in setup_requires.

buildInputs ? []: Build and/or run-time dependencies that need to be be compiled for the host machine. Typically non-Python libraries which are being linked.

checkInputs ? []: Dependencies needed for running the checkPhase. These are added to nativeBuildInputs when doCheck = true. Items listed in tests_require go here.

propagatedBuildInputs ? []: Aside from propagating dependencies, buildPythonPackage also injects code into and wraps executables with the paths included in this list. Items listed in install_requires go here.

File: pkgs/development/tools/gertty/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    requests
+    urwid
+    ply
+    ordereddict
+    alembic
+    GitPython
+    pyyaml
+  ];
+
+  # XXX Looks like all the buildInputs needs to be propagated...
```
Comment (knedlsepp) [inline pkgs/development/tools/gertty/default.nix]: To why see: https://nixos.org/nixpkgs/manual/#reference 15.17.2.2.1.1. buildPythonPackage parameters

File: pkgs/development/tools/gertty/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    GitPython
+    pyyaml
+  ];
+
+  src = python3.pkgs.fetchPypi {
+    inherit pname version;
+    sha256 = "1pvpca59z5rcir7lz2yimwiqjvd8yhnwjskxna2bslyfwv996d8w";
+  };
+
+  doCheck = true;
```
Comment (knedlsepp) [inline pkgs/development/tools/gertty/default.nix]: The logs show that `Ran 0 tests in 0.000s`. This might mean that the the package either doesn't contain any tests or that `python setup.py tests` of the project is not properly set up. in which case some python packages require manual definition of the  `installCheckPhase`. Could you verify which one is the case?

**Review by knedlsepp — CHANGES_REQUESTED** [review]
Congratulations on your first PR in nixpkgs! :+1: 
There's some minor ways on improving this derivation still. A lot of the information on that you can find in https://nixos.org/nixpkgs/manual/#reference I did try to point to the sections that contain the relevant information.

File: pkgs/development/tools/gertty/default.nix [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+  # XXX Missing maintainers?
+  meta = {
+    homepage = "http://ttygroup.org/gertty/index.html";
+    description = "Gertty is a console-based interface to the Gerrit Code Review system.";
+    licence = stdenv.lib.licenses.asl20;
```
Comment (knedlsepp) [inline pkgs/development/tools/gertty/default.nix]: Typo: `license`

File: pkgs/development/tools/gertty/default.nix (hunk 6 above)
Comment (Dav1d23) [reply pkgs/development/tools/gertty/default.nix]: Looks like there are no tests in the repo. The only thing seems to be some style check, which I'm not really performing here then.

File: pkgs/development/tools/gertty/default.nix (hunk 5 above)
Comment (Dav1d23) [reply pkgs/development/tools/gertty/default.nix]: Thanks, I had quite a read :)

File: pkgs/development/tools/gertty/default.nix (hunk 3 above)
Comment (Dav1d23) [reply pkgs/development/tools/gertty/default.nix]: I choose the application, since it looks more what I'm trying to have here.

File: pkgs/development/tools/gertty/default.nix (hunk 2 above)
Comment (Dav1d23) [reply pkgs/development/tools/gertty/default.nix]: I'm on the list :)

File: pkgs/development/tools/gertty/default.nix (hunk 1 above)
Comment (Dav1d23) [reply pkgs/development/tools/gertty/default.nix]: indeed, thanks

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -1667,6 +1667,12 @@
     email = "christoph.senjak@googlemail.com";
     name = "Christoph-Simon Senjak";
   };
+  dav1d23 = {
+    email = "petrodavi@gmail.com";
+    github = "dav1d23";
+    githubId = 10028479;
+    name = "David Petroni";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: this should be a separate commit:

```
maitainers: add dav2d23
```

**Review by jonringer — CHANGES_REQUESTED** [review]
the python package needs to follow python packaging guidelines: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#building-packages-and-applications

**Comment by JohnAZoidberg** [comment]
Hey @Dav1d23 do you still use Nix and care to get this packaged?
We can help you make the necessary changes.

**Comment by Dav1d23** [comment]
Hello,

yes, I still use Nix, but I don't really use this package anymore.

As far as I understand, I only put the derivation in the wrong place, but
now something else might be missing or it may just not work anymore!

So, please go ahead with the needed changes if that's easy for you, I doubt
I will have time to put into this anytime soon, sorry :(
